lib/matplotlib/tests/test_collections.py

```diff
@@ -448,7 +448,6 @@ def test_picking():
 
 def test_linestyle_single_dashes():
     plt.scatter([0, 1, 2], [0, 1, 2], linestyle=(0., [2., 2.]))
-    plt.draw()
 
```

Comment:
I'm confused: in the test, we are in non-interactive mode.  Is a draw executed in this test?  If not, should it be, to complete the smoke-test?

Reply:
good point, should probably replace all of these with `fig.canvas.draw()`